cairo/tests/utils/serde.py

```diff
@@ -286,6 +297,16 @@ def get_offset(self, cairo_type):
             except (ValueError, AttributeError):
                 return 1
 
+    def get_offsets(self, cairo_types: List[CairoType]):
+        """Given a list of Cairo types, return the offset for each type."""
+        offsets = []
+        total = 0
+        for cairo_type in reversed(cairo_types):
+            offset = self.get_offset(cairo_type)
+            total += offset
+            offsets.append(total)
```

Comment:
this is a cumsum over the offset, make this explicit and pythonic, see https://stackoverflow.com/questions/15889131/how-to-find-the-cumulative-sum-of-numbers-in-a-list/15889203#15889203 (itertools or iteration with :=)